src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp

```diff
@@ -530,8 +530,11 @@ void CurlHttpClient::InitGlobalState()
 
 void CurlHttpClient::CleanupGlobalState()
 {
-    curl_global_cleanup();
-    isInit = false;
+    if (isInit)
```

Comment:
is this thread safe? should we put it behing a [unique_lock](https://en.cppreference.com/w/cpp/thread/unique_lock) ? we do things like that [elsewhere](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/ResourceManager.h#L123)

Reply:
Yes. isInit is atomic:
```cpp
std::atomic<bool> CurlHttpClient::isInit(false);
```